janitor/index.d.ts

```diff
@@ -67,11 +67,11 @@ export class Janitor<U extends object | void = void> {
 	 * @returns The new constructed object from the passed `object` class.
 	 */
 	public AddObject<
-		O extends Constructable<unknown, Array<unknown>>,
-		M extends undefined | ((this: O) => void) | ((_: O) => void) | ExtractKeys<O, () => void> | true,
-		I extends keyof U | undefined = undefined,
-		Args extends Array<unknown> = [],
-	>(object: Constructable<O, Args>, methodName?: M, index?: I, ...args: Args): O;
+			T,
+			O extends Constructable<T>,
+			M extends undefined | ExtractKeys<InstanceType<O>, () => void> | true,
```

Comment:
Should we be dropping `((this: O) => void) | ((_: O) => void)` from here?

Though we should be using `T`, not `O` here if we keep it.

Might be more clear if we use `ExtractKeys<T, () => void>` rather than `ExtractKeys<InstanceType<O>, () => void>`

Reply:
Looking at Janitor's internals, we don't need `((this: T) => void) | ((_: T) => void>`:

https://github.com/howmanysmall/Janitor/blob/af930bf8939b7758b23bf62fb651ab5f348cdc52/src/init.luau#L252

we can see that Janitor warns if the method doesn't exist on the created instance, and it will later do nothing. We should explicitly ban this behavior from our consumers.